piptools/scripts/compile.py

```diff
@@ -114,9 +114,10 @@ def cli(verbose, dry_run, pre, rebuild, find_links, index_url, extra_index_url,
         for host in trusted_host:
             pip_args.extend(['--trusted-host', host])
 
-    pip_options, _ = pip_options.parse_args(pip_args)
+    pip_options, _ = pip_command.parse_args(pip_args)
 
-    repository = PyPIRepository(pip_options)
+    session = pip_command._build_session(pip_options)
```

Comment:
I do think using private pip APIs can be a bit problematic, but I can see the usecase here though. Still, relying on these private APIs means at any time in the future pip-tools can break when pip decides to change its internals. We'll see what happens, I guess :)

Thanks for the PR!